compiler/rustc_metadata/src/creader.rs

```diff
@@ -598,7 +598,10 @@ impl<'a> CrateLoader<'a> {
         // don't want to match a host crate against an equivalent target one
         // already loaded.
         let root = library.metadata.get_root();
-        Ok(Some(if locator.triple == self.sess.opts.target_triple {
+        // Proc-macro crates are always compiled for the host target, so they should be reused even if we're cross-compiling.
```

Comment:
It's better to remove this comment or replace it with something like "we don't know what this condition means".
Right now it may look plausible to an unsuspecting reader, but it doesn't really make sense.